google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -928,8 +931,14 @@ public Transaction call() throws Exception {
     }
 
     TransactionContextImpl newTransaction() {
-      TransactionContextImpl txn =
-          new TransactionContextImpl(this, readyTransactionId, gapicRpc, defaultPrefetchChunks);
+      TransactionContextImpl txn;
```

Comment:
Can this be accomplished with a subclass used in a test rather than changing `SpannerImpl`?

Reply:
Good idea, I changed it to an implementation using a subclass. I still needed a couple of small additions to `SpannerImpl`, but those are limited to a couple of package-private getter methods.